assign-doi-datacite/src/main/java/no/unit/nva/doi/datacite/mdsclient/DataCiteMdsConnection.java

```diff
@@ -59,10 +63,10 @@ public DataCiteMdsConnection(HttpClient httpClient, String host, int port) {
     public HttpResponse<String> postMetadata(String doi, String dataCiteXml) throws IOException,
                                                                                     URISyntaxException,
                                                                                     InterruptedException {
-        Objects.requireNonNull(doi);
-        Objects.requireNonNull(dataCiteXml);
+        Objects.requireNonNull(doi, MISSING_DOI_IDENTIFIER_ARGUMENT);
+        Objects.requireNonNull(dataCiteXml, MISSING_DATACITE_XML_ARGUMENT);
```

Comment:
Again, we check only that something is present, not that it is structured in any way, so the error message is misleading.

Should it be parsed now, or later?